core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -1021,10 +1020,9 @@ public void syncPersistDirectory(InodeDirectory dir, JournalContext journalConte
               dir.setOwner(status.getOwner())
                   .setGroup(status.getGroup())
                   .setMode(status.getMode());
-            } catch (FileNotFoundException e) {
-              // Retry creation given that the directory might have just been removed.
-              LOG.warn("Directory {} no longer exists on UFS. Retry creation.", ufsUri);
-              continue;
+            } catch (Exception e) {
+              throw new IOException(String.format("Cannot create or sync UFS directory %s.",
```

Comment:
include the message of the root cause in the detail string - only the detail string is sent over the wire, not the cause chain

Reply:
Done.